Userland/Applications/PixelPaint/Tools/GradientTool.cpp

```diff
@@ -208,6 +250,41 @@ ErrorOr<GUI::Widget*> GradientTool::get_properties_widget()
 
         set_primary_slider(opacity_slider);
 
+        auto hardness_container = TRY(properties_widget->try_add<GUI::Widget>());
+        (void)TRY(hardness_container->try_set_layout<GUI::HorizontalBoxLayout>());
+        hardness_container->set_fixed_height(0);
+
+        mode_combobox->on_change = [this, hardness_container](auto&, auto& model_index) {
+            VERIFY(model_index.row() >= 0);
+            VERIFY(model_index.row() < (int)GradientMode::__Count);
+
+            GradientMode selected_mode = model_index.row() == 0 ? GradientMode::Linear : GradientMode::Radial;
+
+            if (m_mode != selected_mode) {
+                m_mode = selected_mode;
+                reset();
+            }
+
+            if (m_mode == GradientMode::Linear)
+                hardness_container->set_fixed_height(0);
+            else
+                hardness_container->set_fixed_height(20);
```

Comment:
Setting the height to 0 to hide something feels a bit weird. We can do:

```suggestion
            hardness_container->set_visible(m_mode == GradientMode::Radial);
```

And then change line 255 `hardness_container->set_fixed_height(0);` to set the actual height, and set it visible/invisible.

I'm noticing that the size of the properties container doesn't change then, but that's a bug in the layout system.